hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -239,4 +264,53 @@ private Nft buildNewlyMintedNft(
                 .metadata(meta)
                 .build();
     }
+
+    @NonNull
+    @Override
+    public Fees calculateFees(@NonNull final FeeContext feeContext) {
+        final var op = feeContext.body().tokenMintOrThrow();
+        final var readableTokenStore = feeContext.readableStore(ReadableTokenStore.class);
+        final var tokenType = readableTokenStore.get(op.tokenOrThrow()).tokenType();
+        final var subType = tokenType == TokenType.FUNGIBLE_COMMON
+                ? SubType.TOKEN_FUNGIBLE_COMMON
+                : SubType.TOKEN_NON_FUNGIBLE_UNIQUE;
+
+        final var readableAccountStore = feeContext.readableStore(ReadableAccountStore.class);
+        final var payerId = feeContext.body().transactionIDOrThrow().accountIDOrThrow();
+        final var payerKey = readableAccountStore.getAccountById(payerId).keyOrThrow();
+
+        final var calculator = feeContext.feeCalculator(subType);
+        if (SubType.TOKEN_NON_FUNGIBLE_UNIQUE.equals(subType)) {
+            // The price of nft mint should be increased based on number of signatures.
+            // The first signature is free and is accounted in the base price, so we only need to add
+            // the price of the rest of the signatures.
+            calculator.addVerificationsPerTransaction(Math.max(0, numSimpleKeys(payerKey) - 1L));
+        }
+
+        final var meta = TOKEN_OPS_USAGE_UTILS.tokenMintUsageFrom(fromPbj(feeContext.body()), fromPbj(subType), 76000L);
+
+        calculator.addBytesPerTransaction(meta.getBpt());
+        calculator.addRamByteSeconds(meta.getRbs());
+        calculator.addNetworkRamByteSeconds(meta.getTransferRecordDb() * USAGE_PROPERTIES.legacyReceiptStorageSecs());
+        return calculator.calculate();
+    }
+
+    private int numSimpleKeys(final Key key) {
+        final var count = new AtomicInteger(0);
+        if (key.hasThresholdKey()) {
+            final var keys =
+                    key.thresholdKeyOrThrow().keysOrElse(KeyList.DEFAULT).keysOrElse(Collections.emptyList());
+            for (final var k : keys) {
+                numSimpleKeys(k);
+            }
+        } else if (key.hasKeyList()) {
+            final var keys = key.keyListOrElse(KeyList.DEFAULT).keysOrElse(Collections.emptyList());
+            for (final var k : keys) {
+                numSimpleKeys(k);
+            }
+        } else {
+            count.incrementAndGet();
```

Comment:
Was going to leave a comment about how `ContractID` keys don't make sense to count here; then checked [mono equivalent `calculateKeysMetadata()`](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/fee/FeeBuilder.java#L231) and realized this is the behavior we need to emulate. So well played. 😇 

We need a shorthand comment for "not ideal but required for differential testing."

Reply:
Added a comment. Thanks @tinker-michaelj